device/common/include/traccc/finding/device/build_tracks.hpp

```diff
@@ -18,22 +18,28 @@ namespace traccc::device {
 /// we can build a full track by iterating from a tip link backwardly.
 ///
 /// @param[in] globalIndex         The index of the current thread
+/// @param[in] cfg                    Track finding config object
 /// @param[in] measurements_view   Measurements container view
 /// @param[in] seeds_view          Seed container view
 /// @param[in] link_view           Link container view
 /// @param[in] param_to_link_view  Container for param index -> link index
 /// @param[in] tips_view           Tip link container view
 /// @param[out] track_candidates_view  Track candidate container view
+/// @param[out] valid_indices_view   Valid indices meeting criteria
+/// @param[out] n_valid_tracks       The number of valid tracks meeting criteria
 
+template <typename config_t>
```

Comment:
Do we really need to template on the configuration object type? I'd much prefer if the config object could be some concrete type. 🤔

Reply:
I agree. Shall we do this once we remove the scalar_type template from configuration.

Reply:
Yeah, it doesn't have to be cleaned up in this PR. We should just remember it that we'll want to make the configuration types a little easier to use, here as well. :wink: